### PR TITLE
Make sidebar more consistent and add ts

### PR DESCRIPTION
**Pull request by OskarDamkjaer**
Previously the favorites tab didn't have any header:
bild
And the some tabs had different names on hover compared to their headers.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,7 +15,6 @@ dist/
 NOTICE.xml
 *.tgz
 *.tar.gz
-release-notes.md
```
Comment (OskarDamkjaer) [inline .gitignore:18]: this file was removed a while ago

File: package.json [hunk 2]
```diff
@@ -135,7 +137,7 @@
   "dependencies": {
     "@literal-jsx/parser": "^0.1.7",
     "@mdx-js/runtime": "^1.6.1",
-    "@neo4j/browser-lambda-parser": "1.0.4",
+    "@neo4j/browser-lambda-parser": "1.0.5",
```
Comment (OskarDamkjaer) [inline package.json:140]: update needed for unrelated bug fix

File: src/browser/modules/Sidebar/About.tsx [hunk 3]
```diff
@@ -142,19 +158,7 @@ const About = ({ serverVersion, serverEdition }) => (
...
-  }
-  const tokenisedServerVersion = serverVersion && serverVersion.split('.')
-  const releaseTag = tokenisedServerVersion.join('')
-  const urlServerVersion =
-    serverVersion && tokenisedServerVersion.splice(0, 2).join('.')
-  return `https://github.com/neo4j/neo4j/wiki/Neo4j-${urlServerVersion}-changelog#${releaseTag}`
-}
-
-const mapStateToProps = state => {
+const mapStateToProps = (state: any) => {
```
Comment (OskarDamkjaer) [inline src/browser/modules/Sidebar/About.tsx:161]: Getting a better solution to state types didn't seem trivial. It'll be something to look into in the future

File: .gitignore (hunk 1 above)
Comment (oskarhane) [reply .gitignore:18]: IMO: We want to keep ignoring this so CI doesn't have to think about it when it's generated

Comment (OskarDamkjaer) [reply .gitignore:18]: Oh! I didn't know it was generated! I just saw the one that was years old and assumed it's something we no longer did. I'll put it back in

File: package.json (hunk 2 above)
Comment (OskarDamkjaer) [reply package.json:140]: Fixes #1112
